src/Jenssegers/Mongodb/Query/Builder.php

```diff
@@ -1153,6 +1168,15 @@ public function options(array $options)
         return $this;
     }
 
+    protected function session(array $options = [])
+    {
+        if ($session = $this->connection->getSession()) {
```

Comment:
Suggestion to not override an existing session:

```suggestion
        if (!isset($options['session']) && $session = $this->connection->getSession()) {
```